test/models/item_test.rb

```diff
@@ -36,6 +47,15 @@ class ItemTest < ActiveSupport::TestCase
     assert_nil first_item.position
   end
 
+  test "should set completed_at on checking" do
+    item = items(:first)
+    assert_nil item.completed_at
+    item.mark
+
+    item.reload
+    assert_equal item.updated_at.to_a, item.completed_at.to_a
```

Comment:
@sergeyradkevich, I've used it because date comparison needs implementation of '#==' otherwise('No visible difference...' error).